pkg/segment/results/mresults/tests/metricsresults_test.go

```diff
@@ -407,6 +407,62 @@ func TestCalculateInterval(t *testing.T) {
 	}
 }
 
+func Test_GetResults_Modulo(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:    52.5,
+			3600: 23,
+			7200: -10,
+		},
+		map[uint32]float64{
+			0:    2.5,
+			3600: 3,
+			7200: 0,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetModulo,
+				Constant:   5,
+			},
+		},
```

Comment:
Nice test with a negative number. You should add another case to do a negative number not divisible by 5. So maybe `-7 % 5`. Some languages will give 3 and others -2. What does Go give? What should we give?

Reply:
I tried that in promql ui, if the equation is like you said is: -7 % 5, it will return -2